packages/dds/tree/src/simple-tree/api/tree.ts

```diff
@@ -68,6 +66,64 @@ export interface ITree extends IFluidLoadable {
 	): TreeView<TRoot>;
 }
 
+/**
+ * A tree that may be {@link ViewableTree.viewWith | viewed} or {@link BranchableTree.branch | branched}.
+ * @alpha
+ */
+export interface BranchableTree extends ViewableTree {
+	/**
+	 * Create a new branch of the current state of the tree.
+	 * @remarks
+	 * The branch can be read and modified independently of the original tree.
+	 * A {@link TreeView | view} of the branch can be acquired using {@link ViewableTree.viewWith}().
+	 * Edits to the branch do not affect the original tree until they are merged back in.
+	 * Braches may also be rebased onto each other.
+	 */
+	branch(): TreeBranch;
+
+	/**
+	 * Apply all the new changes on the given branch to this tree.
+	 * @param branch - a branch which was created by a call to {@link BranchableTree.branch}().
+	 * @remarks It is automatically disposed after the merge completes.
+	 */
+	merge(branch: TreeBranch): void;
+
+	/**
+	 * Apply all the new changes on the given branch to this tree.
+	 * @param branch - a branch which was created by a call to {@link BranchableTree.branch}().
+	 * @param disposeView - whether or not to dispose `branch` after the merge completes.
+	 */
+	merge(branch: TreeBranch, disposeView: boolean): void;
+
+	/**
+	 * Rebase the changes that have been applied to the given branch over all the new changes in this tree.
+	 * @param branch - a branch which was created by a call to {@link BranchableTree.branch}().
+	 * It is modified by this operation.
+	 */
+	rebase(branch: TreeBranch): void;
+}
+
+/**
+ * Channel for a Fluid Tree DDS.
+ * @remarks
+ * Allows storing and collaboratively editing schema-aware hierarchial data.
+ * @sealed @public
+ */
+export interface ITree extends ViewableTree, IFluidLoadable {}
+
+/**
+ * A {@link BranchableTree.branch | fork} of a tree.
+ * @alpha
+ */
+export interface TreeBranch extends BranchableTree, IDisposable {
+	/**
+	 * Rebase the changes that have been applied to this view over all the new changes in the given view.
+	 * @param fork - Either the root tree or a tree that was created by a call to `fork()`.
```

Comment:
I don't totally follow the param docs here. Isn't this just the _target_ of the rebase operation? I.e. _this_ branch rebased onto `fork`? In which case, doesn't the target get modified?

Reply:
Yes and Yes and No. When you "rebase X over Y" or "rebase X onto Y", Y remains the same and X changes to account for the new content of Y. The other function, `rebase` (not `rebaseOnto`), works the other way. You can think of rebasing X onto Y as "X catching up to Y" - so it makes sense that X is modified whereas Y is not. In fact, the name "rebase" is illuminating as well - when you "rebase X onto Y", it is X that is being "re-based" - X is the object of the verb and it is the branch that is having its "base commit" (the common ancestor commit that it shares with Y) changed to be the head commit of Y.

I'm not really sure how deep to go with the docs here (especially at this time - alpha vs. public). Rebasing is complicated enough that I'm just sort of assuming a user of this knows what they are doing. We could alternatively explain what rebasing is, and why you would use it, etc. etc. in these docs, but I'm not sure if that's our responsibility. There are many other docs out there explaining the concept. I used git for many years before I even remotely understood what a rebase was, and I don't think I'm alone. Probably, this suffices for now, but when we make this public we can have a companion blog post or a markdown file that goes a bit deeper and gives a crash course on rebasing as it fits into our system.

Reply:
Oooh, right. I was thinking about it backwards. That makes sense.

Reply:
Since this is a public-facing API, might still be worth more detail and possibly examples. But since this is currently alpha, that can probably wait.